wandb/sdk/launch/agent/agent.py

```diff
@@ -79,10 +103,21 @@ def __init__(self, api: Api, config: Dict[str, Any]):
         self._name = ""  # hacky: want to display this to the user but we don't get it back from gql until polling starts. fix later
 
     @property
-    def job_ids(self) -> List[Union[int, str]]:
-        """Returns a list of keys running job ids for the agent."""
+    def thread_ids(self) -> List[int]:
+        """Returns a list of keys running thread ids for the agent."""
         return list(self._jobs.keys())
 
+    @property
+    def job_ids(self) -> List[Union[str]]:
+        """Returns a list of keys running job ids for the agent."""
+        job_ids: List[str] = []
+        with self._jobs_lock:
+            for key in list(self._jobs.keys()):
+                run = self._jobs[key].run
+                if run is not None:
+                    job_ids.append(run.id)
```

Comment:
more pythonic: possibly less readable
```suggestion
            for job in self._jobs.values():
                if job.run:
                    job_ids.append(job.run.id)
```

Reply:
actually:
```suggestion
            job_ids = [job.run for job in self._jobs.values() if job.run]
```